c7n/resources/dms.py

```diff
@@ -105,3 +121,114 @@ def process(self, resources):
         client = local_session(self.manager.session_factory).client('dms')
         for arn, r in zip(self.manager.get_arns(resources), resources):
             client.delete_replication_instance(ReplicationInstanceArn=arn)
+
+
+@ReplicationInstance.action_registry.register('tag')
+class InstanceTag(Tag):
+    """
+    Add tag(s) to a replication instance
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: tag-dms-required
+                  resource: dms-instance
+                  filters:
+                    - "tag:RequireTag": absent
+                  actions:
+                    - type: tag
+                      key: RequiredTag
+                      value: RequiredTagValue
+    """
+    permissions = ('dms:AddTagsToResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('dms')
+        tags_list = []
+        for t in tags:
+            tags_list.append({'Key': t['Key'], 'Value': t['Value']})
+        for r in resources:
+            try:
+                client.add_tags_to_resource(
+                    ResourceArn=r['ReplicationInstanceArn'],
+                    Tags=tags_list)
+            except ClientError as e:
+                self.log.exception(
+                    'Exception while adding tags to %s: %s',
+                    r['ReplicationInstanceIdentifier'], e)
+                continue
+
+
+@ReplicationInstance.action_registry.register('remove-tag')
+class InstanceRemoveTag(RemoveTag):
+    """
+    Remove tag(s) from a replication instance
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: delete-single-az-dms
+                  resource: dms-instance
+                  filters:
+                    - "tag:InvalidTag": present
+                  actions:
+                    - type: remove-tag
+                      tags: ["InvalidTag"]
+    """
+    permissions = ('dms:RemoveTagsFromResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('dms')
+        for r in resources:
+            try:
+                client.remove_tags_from_resource(
+                    ResourceArn=r['ReplicationInstanceArn'],
+                    TagKeys=tags)
+            except ClientError as e:
```

Comment:
@joshuaroot check for `ResourceNotFoundFault`, log the error and continue.  Raise for all other errors.